### PR TITLE
Added `ansible_python_interpreter` to Debian.yml vars file

### LINKED ISSUE #606: `Install - IPA client test` seems to require `six` module for Python3 but it isn't installed
Hi everyone.

I am trying to apply the `ipaclient` role to a Debian buster node which has Python 3.7.3 installed (package `python3` from default Debian sources).
In `plugins/modules/ipaclient_test.py` there is in lines 195:198 something that looks like a Python2/Python3 differentiation:

```python
try:
    from six.moves.configparser import RawConfigParser
except ImportError:
    from ConfigParser import RawConfigParser
```

So: Since `ConfigParser` has been renamed to `configparser` in Python3, if `six` is not installed, neither `six.moves.configparser.RawConfigParser`, nor `ConfigParser.RawConfigParser` is found.
I also can't find this requirement mentioned anywhere outside of the code.

`ipaclient_install_packages` is `true`; so I would have expected that `python3-six` is installed along with the other requirements for Debian.

### LINKED ISSUE #607: No module named 'ipapython'
Hi everyone.

I am trying to apply the ipaclient role to a Debian buster node which has Python 3.7.3 installed (package python3 from default Debian sources).
I installed `freeipa.ansible_freeipa` from Ansible Galaxy and my boring Playbook looks like this:

```YAML
---
- name: Playbook to configure IPA clients with username/password
  hosts: ipaclients
  become: true

  roles:
    - role: freeipa.ansible_freeipa.ipaclient
      state: present
```

So, not much space for mistakes, I'd say ;-)

It is applied to a node with these parameters set:

```
$ ansible-inventory --host idtestlxc
{
    "ansible_host": "idtestlxc.our-dom.cloud",
    "ipaadmin_password": "{{ vault_ipaadmin_password }}",
    "ipaadmin_principal": "{{ vault_ipaadmin_principal }}",
    "ipaclient_install_packages": true,
    "ipaclient_servers": "id1.our-dom.cloud,id2.our-dom.cloud",
    "ipaclient_use_otp": true,
    "ipadm_password": "{{ vault_ipadm_password }}",
    "ipaserver_auto_forwarders": false,
    "ipaserver_domain": "id.our-dom.cloud",
    "ipaserver_realm": "ID.OUR-DOM.CLOUD",
    "ipaserver_setup_dns": true,
    "vault_ipaadmin_password": "XXXXXXXXXXXXXXX",
    "vault_ipaadmin_principal": "mr.x",
    "vault_ipadm_password": "XXXXXXXXXXXXXXX"
}
```
I guess the most important part here is `ipaclient_install_packages`, which is set to `true`.

`ipapython` seems to be available in PyPi for Debian, only. But I'm unsure what is the correct path from here: Why do I face this issue even with `ipaclient_install_packages` set to `true`; my understanding here is that this basically says "_please take care for the dependencies of the module for me_", which seems to not be happening in this case.
Should I install `ipaclient_install_packages` from PyPi myself now? If so, it would be nice to read about this in the Prerequisites docs section then rather to find this out on a "run, debug, repeat" based approach.

PS: Sorry, I seem to have forgotten to paste the issue:

```
TASK [freeipa.ansible_freeipa.ipaclient : Install - IPA client test] ****************************************************************************************************************************
...
The full traceback is:
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-tmp-1627376833.765989-16986-193057600277786/AnsiballZ_ipaclient_test.py", line 100, in <module>
    _ansiballz_main()
  File "/root/.ansible/tmp/ansible-tmp-1627376833.765989-16986-193057600277786/AnsiballZ_ipaclient_test.py", line 92, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/root/.ansible/tmp/ansible-tmp-1627376833.765989-16986-193057600277786/AnsiballZ_ipaclient_test.py", line 41, in invoke_module
    run_name='__main__', alter_sys=True)
  File "/usr/lib/python3.7/runpy.py", line 205, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_freeipa.ansible_freeipa.ipaclient_test_payload_zypnqyjj/ansible_freeipa.ansible_freeipa.ipaclient_test_payload.zip/ansible_collections/freeipa/ansible_freeipa/plugins/modules/ipaclient_test.py", line 201, in <module>
  File "/tmp/ansible_freeipa.ansible_freeipa.ipaclient_test_payload_zypnqyjj/ansible_freeipa.ansible_freeipa.ipaclient_test_payload.zip/ansible_collections/freeipa/ansible_freeipa/plugins/module_utils/ansible_ipa_client.py", line 57, in <module>
ModuleNotFoundError: No module named 'ipapython'
fatal: [idtestlxc]: FAILED! => {
    "changed": false,
    "module_stderr": "Shared connection to idtestlxc.my-dom.cloud closed.\r\n",
    "module_stdout": "Traceback (most recent call last):\r\n  File \"/root/.ansible/tmp/ansible-tmp-1627376833.765989-16986-193057600277786/AnsiballZ_ipaclient_test.py\", line 100, in <module>\r\n    _ansiballz_main()\r\n  File \"/root/.ansible/tmp/ansible-tmp-1627376833.765989-16986-193057600277786/AnsiballZ_ipaclient_test.py\", line 92, in _ansiballz_main\r\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\r\n  File \"/root/.ansible/tmp/ansible-tmp-1627376833.765989-16986-193057600277786/AnsiballZ_ipaclient_test.py\", line 41, in invoke_module\r\n    run_name='__main__', alter_sys=True)\r\n  File \"/usr/lib/python3.7/runpy.py\", line 205, in run_module\r\n    return _run_module_code(code, init_globals, run_name, mod_spec)\r\n  File \"/usr/lib/python3.7/runpy.py\", line 96, in _run_module_code\r\n    mod_name, mod_spec, pkg_name, script_name)\r\n  File \"/usr/lib/python3.7/runpy.py\", line 85, in _run_code\r\n    exec(code, run_globals)\r\n  File \"/tmp/ansible_freeipa.ansible_freeipa.ipaclient_test_payload_zypnqyjj/ansible_freeipa.ansible_freeipa.ipaclient_test_payload.zip/ansible_collections/freeipa/ansible_freeipa/plugins/modules/ipaclient_test.py\", line 201, in <module>\r\n  File \"/tmp/ansible_freeipa.ansible_freeipa.ipaclient_test_payload_zypnqyjj/ansible_freeipa.ansible_freeipa.ipaclient_test_payload.zip/ansible_collections/freeipa/ansible_freeipa/plugins/module_utils/ansible_ipa_client.py\", line 57, in <module>\r\nModuleNotFoundError: No module named 'ipapython'\r\n",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
```

### PR DESCRIPTION
**Pull request by The-Judge**
Added `ansible_python_interpreter` to Debian.yml vars (as it is in molecule file at `molecule/resources/playbooks/roles/ipaclient/vars/Debian.yml`) to fix #607 , fix #606

### CODE REVIEW

**Comment by The-Judge** [comment]
Fixes #606

File: roles/ipaclient/vars/Debian-buster.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# vars/Debian.yml
+ipaclient_packages: [ "freeipa-client" ]
+ansible_python_interpreter: '/usr/bin/python2'
```
Comment (t-woerner) [inline roles/ipaclient/vars/Debian-buster.yml:3]: Is this really needed for all Debian (based) releases?

Comment (The-Judge) [reply roles/ipaclient/vars/Debian-buster.yml:3]: At least it seems to be for Debian buster, which this file is named for for the `with_first_found` conditional include statement.

**Comment by t-woerner** [comment]
@rjeffman What do you think about adding `    - "{{ role_path }}/vars/{{ ansible_facts['distribution'] }}-{{ ansible_facts['ansible_distribution_release'] }}.yml"` for all roles?

**Review by rjeffman — CHANGES_REQUESTED** [review]
Here's what I suggest:
* Undo the changes to `roles/ipaclient/tasks/main.yml`
* Rename `roles/ipaclient/vars/Debian-buster.yml` to `roles/ipaclient/vars/Debian-10.yml`
* Add a comment before setting `ansible_python_interpreter` saying something like:
> # On Debian Buster freeipa-client package requires Python2 packages and must use python2 as Python interpreter.

This way, we are making minimal changes, providing enough to support Debian Buster, and changing only distro-specific files.

Thank you for finding a fix for this issue!

**Comment by t-woerner** [comment]
Can we close this after #627 has been merged?

**Comment by The-Judge** [comment]
Since it's quite common for Debian to name the OS version in both terms (some prefer the codename e.g. "buster" or "bullseye", others prefer the numeric name "10" or "11"), I added this alternative to `roles/ipaclient/tasks/main.yml`, so users can choose.
Anyways, be gone ;)

**Review by rjeffman — CHANGES_REQUESTED** [review]
@The-Judge, please join all the commits into a single one, and remove the `.idea/.gitignore` file.

**Comment by rjeffman** [comment]
Thank you for your contribution.

The changes proposed have been incorporated to PR #699, which have been merged, so closing this one.

**Pull request closed without merging**